tests/system_tests/test_web_frameworks.py

```diff
@@ -258,6 +262,9 @@ def run_test_with_handled_exception(self, debug_info, framework,
             web_client_thread.start()
             web_client_thread.join(timeout=1)
 
+            # Give enough time for all events to be received
+            time.sleep(3)
```

Comment:
is there no way to query the events received and wait in a loop.
Might be better to do so, as I've done something similar in the past and have noticed such tests to fail occasionally (possibly when server is under load).

Reply:
I will change how we handle this.

Reply:
Not mandatory, but good if we could

Reply:
Created this #755 to tackle the problem with web frame work tests.